packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -122,7 +122,23 @@ abstract class FlutterCommand extends Command<Null> {
     _usesPubOption = true;
   }
 
-  void addBuildModeFlags({ bool defaultToRelease: true }) {
+  void usesBuildNumberOption() {
+    argParser.addOption('buildNumber',
+        help: 'An integer used as an internal version number.\n'
+            'On Android it is used as \'versionCode\'\n'
+            'On xcode builds it is used as \'CFBundleVersion\'',
+        valueHelp: '42');
+  }
+
+  void usesBuildNameOption() {
+    argParser.addOption('buildName',
+        help: 'A string used as the version number shown to users.\n'
+            'On Android it is used as \'versionName\'\n'
+            'On xcode builds it is used as \'CFBundleShortVersionString\'',
```

Comment:
I would add formatting expectations in the help too. e.g. on android, it could be anything and ios expects x.y.z

Reply:
Android and iOS can have different version names but to have it consistent I changed it to: `A "x.y.z" string used as the version number shown to users.`